### ISSUE TITLE
Can not reproduce this issue

### ISSUE THREAD

**Issue by zyznull** [voidism/DiffCSE#8]
I have used this run_diffcse.sh, but the avg sts result is only 77.00. Is there some thing need to notice?

**Comment by Jhansoll** [comment]
I had the same issue.

**Comment by Gitsamshi** [comment]
Same here, got the result of 77.33.

**Comment by yiren-jian** [comment]
I got 77.44. I believe the hyper-parameters are very sensitive to the device you use (learn this from my own projects). You probably need to re-search hyper-parameters to get the same results.

My own work on CSE tells me that: the results can be sensitive to python version, GPU device, cuda version, pytorch version and huggingface version.

**Comment by hbin0701** [comment]
Me too.

**Comment by qmin2** [comment]
Here also :(

![image](https://user-images.githubusercontent.com/78135234/185291629-ada84257-af9a-4a58-9f72-15e93426e186.png)

**Comment by voidism** [comment]
Hi all!

After I tried some experiments on another machine, I found that the hyperparams are very sensitive to the device you use. I cannot reproduce the results on another machine with the same hyperparams either. Your python/pytorch/cuda/huggingface version will affect your results. The hyperparams on the paper are only suitable for the first machine I used, so you probably need to re-search hyperparams to get the same results on your machine.

A recent CSE paper https://github.com/yiren-jian/NonLing-CSE/tree/main/VisualCSE seems also have this problem for CSE. They suggest use the same hardwares and software versions to faithfully reproduce the results in the paper.
